common/src/chain/transaction/transaction_index.rs

```diff
@@ -1,10 +1,33 @@
-use crate::primitives::H256;
+use crate::{
+    chain::block::Block,
+    primitives::{Id, H256},
+};
 use parity_scale_codec_derive::{Decode, Encode};
 
-#[derive(Clone, Copy, Debug, PartialEq, Eq, Encode, Decode)]
+use super::Transaction;
+
+#[derive(Clone, Debug, PartialEq, Eq, Encode, Decode)]
+pub enum Spender {
+    RegularInput(Id<Transaction>),
+    StakeKernel(Id<Block>),
+}
```

Comment:
What's the difference between this and `OutPointSourceId`? Could these be one type?

Reply:
While they look the same, the difference is very subtle. The question related here "is every outpoint type equivalent to every possible input type"? The answer is that outpoint types are a superset of input types, because proof of work, for example, doesn't have inputs (like stake kernels). So it's a design decision whether we ant to distinguish between those. I'm open to suggestions here.